kwave/utils/kwave_array.py

```diff
@@ -83,6 +90,57 @@ def __post_init__(self):
 
         self.measure = float(self.measure)
 
+    def is_close(self, other: "Element", rtol=1e-05, atol=1e-08, equal_nan=False):
+        """Compares 2 Elements to a certain precision for their numerical
+        fields. This differs from the __eq__ method which requires a perfect
+        match between numerical fields. Works with ints, floats and numpy
+        arrays.
+
+        See numpy.allclose for more details.
+
+        :param other: an instance of Element
+        :param rtol: The relative tolerance parameter, defaults to 1e-05
+        :param atol: The absolute tolerance parameter, defaults to 1e-08
+        :param equal_nan: Whether to compare NaN's as equal, defaults to False
+        :raises ValueError: when other object is not an instance of Element
+        :return: bool
+        """
+        if not isinstance(other, type(self)):
+            raise TypeError(f"{other} with {type(other)} is not of type Element")
+
+        for field in fields(self):
+            self_attr = getattr(self, field.name)
+            other_attr = getattr(other, field.name)
+            if isinstance(self_attr, (float, np.ndarray)):
+                if not np.allclose(self_attr, other_attr, rtol=rtol, atol=atol, equal_nan=equal_nan):
+                    return False
+            else:
+                if self_attr != other_attr:
+                    return False
+        return True
+
+    def __eq__(self, other):
+        """Equality operator that handles all fields but specifically numpy
+        arrays.
+
+        :param other: an instance of Element
+        :raises ValueError: when other object is not an instance of Element
+        :return: bool
+        """
+        if not isinstance(other, type(self)):
+            raise TypeError(f"{other} with {type(other)} is not of type Element")
+
+        for field in fields(self):
+            self_attr = getattr(self, field.name)
+            other_attr = getattr(other, field.name)
+            if isinstance(self_attr, np.ndarray):
+                if not np.array_equal(self_attr, other_attr):
+                    return False
+            else:
+                if self_attr != other_attr:
+                    return False
+        return True
```

Comment:
please move __eq__ next to __post_init__ to keep the dunder methods together.

Reply:
Will do.